bin/cheritest/cheritest_vm.c

```diff
@@ -602,3 +600,59 @@ cheritest_vm_cow_write(const struct cheri_test *ctp __unused)
 	CHERITEST_CHECK_SYSCALL(close(fd));
 	cheritest_success();
 }
+
+#ifdef __mips__
+
+/*
+ * Test a mapped and unmapped invocation of cloadtags
+ */
+
+void
+test_cloadtags_mapped(const struct cheri_test *ctp __unused)
+{
+	uint64_t tags;
+	void * __capability * __capability c;
+	void * __capability * p;
+
+	p = CHERITEST_CHECK_SYSCALL(mmap(0, PAGE_SIZE, PROT_READ | PROT_WRITE,
+						MAP_ANON, -1, 0));
+	c = (__cheri_tocap void * __capability * __capability) p;
+
+	p[1] = c;
+	p[2] = c;
+
+	__asm__ ("cloadtags %[tags], %[c]"
+		: [tags]"=r"(tags)
+		: [c]"C"(c)
+		: "memory" );
+
+	CHERITEST_VERIFY2(tags == 0x6,
+		"incorrect result from cloadtags, 0x%" PRIx64 "\n", tags);
+
+	munmap(p, PAGE_SIZE);
+	cheritest_success();
+}
+
+void
+test_fault_cloadtags_unmapped(const struct cheri_test *ctp __unused)
+{
+	uint64_t tags = 0;
+	void * __capability c;
+	void * p;
+
+	p = CHERITEST_CHECK_SYSCALL(mmap(0, PAGE_SIZE, PROT_READ | PROT_WRITE,
+						MAP_ANON, -1, 0));
```

Comment:
Indentation here is strange (and not because of diff markers, I checked in vim).